sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -237,13 +237,19 @@ class Dataset[T] private[sql](
   private[sql] def showString(
       _numRows: Int, truncate: Int = 20, vertical: Boolean = false): String = {
     val numRows = _numRows.max(0).min(Int.MaxValue - 1)
-    val takeResult = toDF().take(numRows + 1)
+    val newDf = toDF()
+    val castExprs = newDf.schema.map { f => f.dataType match {
+      // Since binary types in top-level schema fields have a specific format to print,
+      // so we do not cast them to strings here.
+      case BinaryType => s"${f.name}"
```

Comment:
Do we need to surround `f.name` with `s""`? Or we need to add \` around `f.name`?

Reply:
oops, I forgot `